client/signup/steps/website-content/index.tsx

```diff
@@ -213,28 +213,34 @@ export default function WrapperWebsiteContent(
 	);
 	const siteId = useSelector( ( state ) => getSiteId( state, queryObject.siteSlug as string ) );
 
-	const urlParams = new URLSearchParams( window.location.search );
-	const isFromCheckout = 'purchase-success' === urlParams.get( 'notice' );
-	// If the user is coming from checkout, we can wait for a longer period to
-	// ensure that the WoA sync has completed.
-	const maxTries = isFromCheckout ? 30 : 10;
-
-	const { hasValidPurchase, pageTitles, isPollingInProgress } = usePollSiteForDIFMDetails(
-		siteId,
-		maxTries
-	);
+	const [ pageTitles, setPageTitles ] = useState< PageId[] >( [] );
+	const [ isLoading, setIsLoading ] = useState( true );
 
 	useEffect( () => {
-		if ( isPollingInProgress ) {
-			return;
-		}
-		if ( ! hasValidPurchase ) {
-			debug( 'Website content content already submitted, redirecting to home' );
-			page( `/home/${ queryObject.siteSlug }` );
+		async function fetchSelectedPageTitles() {
+			try {
+				const response: { selected_page_titles: PageId[]; is_website_content_submitted: boolean } =
+					await wpcom.req.get( {
+						path: `/sites/${ queryObject.siteSlug }/do-it-for-me/website-content`,
+						apiNamespace: 'wpcom/v2',
+					} );
+
+				setIsLoading( false );
+				setPageTitles( response.selected_page_titles );
+
+				if ( response.is_website_content_submitted ) {
+					debug( 'Website content content already submitted, redirecting to home' );
+					page( `/home/${ queryObject.siteSlug }` );
+				}
+			} catch ( error ) {
+				setIsLoading( false );
```

Comment:
Yes, this was intentional to show the error screen. If the endpoint returns an error, `isLoading` will be `false`, and `pageTitles` will be set to its default value (`[]`). So, this would show the error screen:
<img width="1920" alt="image" src="https://user-images.githubusercontent.com/5436027/199164670-b142292b-51e2-4d1a-9253-22795bb982fe.png">

Reply:
Maybe we can log this error to logstash, but not essential :).